libraries/joomla/database/driver/mysqli.php

```diff
@@ -557,8 +557,7 @@ public function execute()
 
 		if (!is_object($this->connection))
 		{
-			JLog::add(JText::sprintf('JLIB_DATABASE_QUERY_FAILED', $this->errorNum, $this->errorMsg), JLog::ERROR, 'database');
-			throw new RuntimeException($this->errorMsg, $this->errorNum);
+			throw new RuntimeException('Could not connect to MySQL.');
```

Comment:
I'm pretty sure the variables there are empty because connect just throws exceptions when things go wrong :(

So the only tangible information is 'Database Query Failed' which is just as generic (if not more generic) than the connection failed :/